docs/ilab-model-engine.md

```diff
@@ -0,0 +1,201 @@
+
+Open source container tools like [podman](https://podman.io/) and [docker](https://www.docker.com/) are commonly viewed as engines. `ilab` should be a model engine, managing the deployment and creation of AI models. To be clear, I will be using tools like podman as a structural analogy here with full understanding they have different implications and use-cases.
+
+## Key Component
+
+### InstructLab Structure Redesign
+
+Here are a few outlines. The first two that represent the state of `ilab` before/after this enhancement and the third describes `ilab` after I introduce new key commands to this structure. I am including the second outline for clarity purposes, but this EP only covers the contents of the first structure.
+
+```console
+ilab
+|
+|_____chat
+|_____check
+|_____convert
+|_____download
+|_____generate
+|_____init
+|_____list
+|_____serve
+|_____test
+|_____train
+```
+
+vs. after:
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |____serve   (-i)
+|       |____chat
+|       |____evaluate
+|       |
+|_______data
+|       |
+|       |____generate
+|       |
+|_______config
+|       |
+|       |____init
+|       |
+|_______tax
+|       |
+|       |____diff
+|       |____check
+|       |____download
+```
+
+The main point of this structure is to introduce a group hierarchy and to improve the usability of the existing `ilab` structure. Some key things missing in `ilab` currently are:
+
+1. `ilab init` does too many things.
+    - We don't want "black box" commands. This creates bad UX and also ties too many processes together in which one might eventually need to be changed.
+2. ambiguity in "what" we are generating, training etc.
+    - The current structure of `ilab` requires an in depth knowledge of the ordering of commands and background knowledge for how our CLI is built. This is not the best way to grow. Commands that are not intuitive and require a deep dive into the docs just to get started might not engage users as we hope to. Having an intuitive structure in which people can see easily from the docstring what each commands does and how they might use them, makes more sense. We are generating data, then training a model, running a server to interact with that model, then finally chatting with the model
+3. Some commands like `ilab test` and `ilab list` should be renamed to show their true purpose (`ilab model evaluate`and `ilab taxonomy diff`)
+4. We will hit a point where we want to add more commands to `ilab` due to expected integration with the backend, model registries etc. This list will grow to a point where this form of top group organization is not maintainable.
```

Comment:
@cdoern these are great points and excellent justification as to why this change is needed

Reply:
thanks!